homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -1,102 +1,97 @@
 """Support for Waze travel time sensor."""
+import asyncio
 from datetime import timedelta
 import logging
 import re
+from typing import Any, Callable, Dict, List, Optional
 
 import WazeRouteCalculator
-import voluptuous as vol
 
 from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import (
     ATTR_ATTRIBUTION,
     ATTR_LATITUDE,
     ATTR_LONGITUDE,
     CONF_NAME,
     CONF_REGION,
     CONF_UNIT_SYSTEM_IMPERIAL,
-    CONF_UNIT_SYSTEM_METRIC,
     EVENT_HOMEASSISTANT_START,
     TIME_MINUTES,
 )
+from homeassistant.core import HomeAssistant
 from homeassistant.helpers import location
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
 
-_LOGGER = logging.getLogger(__name__)
-
-ATTR_DESTINATION = "destination"
-ATTR_DURATION = "duration"
-ATTR_DISTANCE = "distance"
-ATTR_ORIGIN = "origin"
-ATTR_ROUTE = "route"
-
-ATTRIBUTION = "Powered by Waze"
+from .const import (
+    ATTR_DESTINATION,
+    ATTR_DISTANCE,
+    ATTR_DURATION,
+    ATTR_ORIGIN,
+    ATTR_ROUTE,
+    ATTRIBUTION,
+    CONF_AVOID_FERRIES,
+    CONF_AVOID_SUBSCRIPTION_ROADS,
+    CONF_AVOID_TOLL_ROADS,
+    CONF_DESTINATION,
+    CONF_EXCL_FILTER,
+    CONF_INCL_FILTER,
+    CONF_ORIGIN,
+    CONF_REALTIME,
+    CONF_UNITS,
+    CONF_VEHICLE_TYPE,
+    DOMAIN,
+    ICON,
+    WAZE_SCHEMA,
+)
 
-CONF_DESTINATION = "destination"
-CONF_ORIGIN = "origin"
-CONF_INCL_FILTER = "incl_filter"
-CONF_EXCL_FILTER = "excl_filter"
-CONF_REALTIME = "realtime"
-CONF_UNITS = "units"
-CONF_VEHICLE_TYPE = "vehicle_type"
-CONF_AVOID_TOLL_ROADS = "avoid_toll_roads"
-CONF_AVOID_SUBSCRIPTION_ROADS = "avoid_subscription_roads"
-CONF_AVOID_FERRIES = "avoid_ferries"
+_LOGGER = logging.getLogger(__name__)
 
-DEFAULT_NAME = "Waze Travel Time"
-DEFAULT_REALTIME = True
-DEFAULT_VEHICLE_TYPE = "car"
-DEFAULT_AVOID_TOLL_ROADS = False
-DEFAULT_AVOID_SUBSCRIPTION_ROADS = False
-DEFAULT_AVOID_FERRIES = False
+SCAN_INTERVAL = timedelta(minutes=5)
 
-ICON = "mdi:car"
 
-UNITS = [CONF_UNIT_SYSTEM_METRIC, CONF_UNIT_SYSTEM_IMPERIAL]
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(WAZE_SCHEMA)
 
-REGIONS = ["US", "NA", "EU", "IL", "AU"]
-VEHICLE_TYPES = ["car", "taxi", "motorcycle"]
 
-SCAN_INTERVAL = timedelta(minutes=5)
+def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
Can we make this `async_setup_platform` and avoid the `asyncio.run_coroutine_threadsafe`?